lib/h3core.js

```diff
@@ -245,12 +245,15 @@ function destroyGeoPolygon(geoPolygon) {
     const geofenceOffset = 0;
     const numHolesOffset = geofenceOffset + SZ_GEOFENCE;
     const holesOffset = numHolesOffset + SZ_INT;
-    // Free the outer loop
-    C._free(C.getValue(geoPolygon + geofenceOffset, 'i8*'));
-    // Free the holes, if any
+    // Offset of the geofence vertex array pointer within the Geofence struct
+    const geofenceArrayOffset = SZ_INT;
+    // Free the outer vertex array
+    C._free(C.getValue(geoPolygon + geofenceOffset + geofenceArrayOffset, 'i8*'));
+    // Free the vertex array for the holes, if any
     const numHoles = C.getValue(geoPolygon + numHolesOffset, 'i32');
+    const holes = C.getValue(geoPolygon + holesOffset, 'i32');
     for (let i = 0; i < numHoles; i++) {
-        C._free(C.getValue(geoPolygon + holesOffset + SZ_GEOFENCE * i, 'i8*'));
+        C._free(C.getValue(holes + SZ_GEOFENCE * i + geofenceArrayOffset, 'i8*'));
```

Comment:
I couldn't find a test case that showed the issue, but this was pretty incorrect - we weren't looking up the holes in the right way at all.

Do I need to free `holes` here too? It seems like I might...

Reply:
Updates to free `holes`

Reply:
Yeah, you need to free the holes, [they're a pointer in the GeoPolygon struct](https://github.com/uber/h3/blob/master/src/h3lib/include/h3api.h.in#L104-L111)